turn_alerts/views.py

```diff
@@ -0,0 +1,128 @@
+import re
+
+from django.http import JsonResponse
+from prometheus_client import Counter
+from rest_framework import generics, permissions, status
+from rest_framework.response import Response
+
+from sidekick.models import Organization
+from turn_alerts.serializers import TurnOutboundSerializer, WhatsAppWebhookSerializer
+
+from .models import TurnActions
+from .tasks import start_turn_journey
+
+message_requests_total = Counter(
+    "turn_alerts_message_requests_total",
+    "Total number of message requests",
+    [
+        "fallback_channel",
+        "direction",
+        "message_type",
+    ],
+)
+
+
+event_count = Counter(
+    "turn_alerts_event_count",
+    "Number of events",
+    ["message_status", "error_code", "conversation_id", "conversation_type"],
+)
+
+
+class TurnAlertsLayerView(generics.GenericAPIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            webhook_type = request.headers["X-Turn-Hook-Subscription"]
+        except KeyError:
+            return Response(
+                {"X-Turn-Hook-Subscription": ["This header is required."]},
+                status.HTTP_400_BAD_REQUEST,
+            )
+        org_id = kwargs["org_id"]
+
+        try:
+            organization = Organization.objects.get(id=org_id)
+            turn_alerts = TurnActions.objects.filter(org=organization)
+        except Organization.DoesNotExist:
+            return JsonResponse(data={}, status=status.HTTP_400_BAD_REQUEST)
+
+        on_fallback_channel = request.headers.get("X-Turn-Fallback-Channel", "0") == "1"
+        is_turn_event = request.headers.get("X-Turn-Event", "0") == "1"
+
+        for alert in turn_alerts:
```

Comment:
How will this work if we have multiple alerts configured?

Reply:
I'm not clear on the question. I ahve at least two alerts configured locally for two separate orgs that I've been testing with.

Reply:
Yes [here](https://github.com/praekeltfoundation/rp-sidekick/pull/203/files#diff-b0d1c28a57dc17af29532ba47605fe846d021f96c0b8f9ff01d1a4e3f0a0ce31R55) you are filtering by organisation, so 1 per org will only return 1.

If you have 2 on one org, `journey_id`, `org_error_code`, `engage_token` and `engage_url` gets overwritten for each alert found.

I would rather add the error code and journey id to a dict, which you can then check in the loop where we get the error code. The engage_* doesn't have to be set in the alert loop since we're only dealing with one org at a time.